File: LeetCode/03-longest-substring-without-repeating-chars/solution.cs
public class Solution
{
    public static int LengthOfLongestSubstring(string s) {
        if (s.Length == 0) return 0;
        Dictionary<char, int> charIndex = [];
        int maxLength = 0;
        int left = 0;
        for (int right = 0; right < s.Length; ++right) {
            if (charIndex.ContainsKey(s[right]))
            {
                left = Math.Max(left, charIndex[s[right]] + 1);
            }
            charIndex[s[right]] = right;
            maxLength = Math.Max(maxLength, right - left + 1);
        }
        return maxLength;
    }
}

File: LeetCode/03-longest-substring-without-repeating-chars/solution.java
import java.util.HashMap;
import java.util.Map;

public class solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.isEmpty()) return 0;
        Map<Character, Integer> charIndex = new HashMap<Character, Integer>();
        int maxLength = 0;
        int left = 0;
        for (int right = 0; right < s.length(); ++right) {
            char current = s.charAt(right);
            if (charIndex.containsKey(current)) {
                left = Math.max(left, charIndex.get(current) + 1);
            }
            charIndex.put(current, right);
            maxLength = Math.max(maxLength, right - left + 1);
        }
        return maxLength;
    }

    public static void main(String[] args) {
        
    }
}